### ISSUE TITLE
[BUG]: fetch every "x" minutes breaks if you focus the app the moment it updates

### ISSUE THREAD

**Issue by Owyn** [martinrotter/rssguard#1282]
### Brief description of the issue

![image](https://github.com/martinrotter/rssguard/assets/1309656/1e0f2fff-a3ef-41c8-9022-e56eb7fcc8f7)

I've set these settings yesterday,
ran the program this morning,
and just noticed it has never checked anything 💀

### How to reproduce the bug?

disable "fetch all articles on startup"

### What was the expected result?

periodic fetch working

### What actually happened?

periodic fetch not working

### Debug log

[rssguard.log](https://github.com/martinrotter/rssguard/files/14024481/rssguard.log)

### Operating system and version

* OS: Win 10
* RSS Guard version: 4.3.3

**Comment by martinrotter** [comment]
I see in log:

> time="    10.082" type="debug" -> core: Delaying scheduled feed auto-download for some time since window is focused and updates while focused are disabled by the user and all account caches are empty.

You have setting "Only auto-fetch articles if application is unfocused." - perhaps this is the problem as I tested locally here and it works for me.

**Comment by Owyn** [comment]
> I see in log:
> 
> > time="    10.082" type="debug" -> core: Delaying scheduled feed auto-download for some time since window is focused and updates while focused are disabled by the user and all account caches are empty.
> 
> You have setting "Only auto-fetch articles if application is unfocused." - perhaps this is the problem as I tested locally here and it works for me.

I definitely didn't have it focused all this time, I have set it to minimized (or behind other windows?), but not into the tray icon

**Comment by martinrotter** [comment]
I see you have 15 minutes interval. Are you trying to say that your application did have focus exaclty 15 minutes after running to that auto-fetchin could start?

**Comment by martinrotter** [comment]
Testing.

**Comment by martinrotter** [comment]
Also, in source code, there is really no connection between the 2 features you mention.

**Comment by Owyn** [comment]
I suspected that regular intervals only started after the first fetch,
but now I have tested this on the devbuild and it seems to be working fine...
no idea why it wasn't fetching anything that whole day (even bad feeds weren't marked as red - so they weren't checked 100%) but this isn't the reason it seems 🤷‍♀️

**Comment by martinrotter** [comment]
I see. I close this and once you find the culprit, we can open another ticket for it.

**Comment by Owyn** [comment]
look at the time before the next update
![image](https://github.com/martinrotter/rssguard/assets/1309656/c7a07cc9-f04c-4394-ac9c-1d770f9a2dc8)

The culprit might be that I have opened the RSSguard but I haven't focused it even once and now when I've checked it - it was just lazying around it seems :D

**Comment by Owyn** [comment]
Yes, the lack of initial focus must be the case, reproduced it again: 
![image](https://github.com/martinrotter/rssguard/assets/1309656/69c2a656-5e37-4625-802c-e43efe327185)
(if I alt-tab of it and back - it starts fetching right away)

**Comment by Owyn** [comment]
how it happens (how to reproduce):
- I have rssguard pinned to my taskbar along with other 4 apps I launch after turning on my PC, so first I click rssguard to launch it then I immediately click all other apps to launch it, so rssguard doesn't get any focus this way and without initial fetch - bugs out 
![image](https://github.com/martinrotter/rssguard/assets/1309656/8c85c244-9a6b-41b0-802c-90bb9a325361)

**Comment by Owyn** [comment]
so, I have noticed that initial fetch worked fine but now after half a day - it all broke again and the "next interval" somehow turned into "not a number" or zero or smth, cuz it shows `%n` to the next update again, my speculation is that it happens if you focus the app the moment it updates and so it can't write the "next interval for update" variable cuz the condition of "not in focus" is not satisfied anymore

![Image](https://github.com/user-attachments/assets/0460c6aa-beee-4756-a9b2-8dd0b1c41d7b)

**Comment by Owyn** [comment]
it seems that the update process itself gets broken midway and forever stuck somewhere in logic pathways so that's why there is no "next update" - cuz the current one isn't finished yet (and never will), see how both "fetch now" and "edit" are unavailable

![Image](https://github.com/user-attachments/assets/14e3349e-c01b-48cc-8f0f-348603c24c97)

**Comment by Owyn** [comment]
I usually launch RSS Guard, then 3 other programs, one of them needing UAC confirmation so it also might be stealing a lot of focus

**Comment by martinrotter** [comment]
MANAGED to reproduce.

**Comment by martinrotter** [comment]
OK it is upstream bug, reporting.

**Comment by martinrotter** [comment]
https://bugreports.qt.io/browse/QTBUG-134212

**Comment by martinrotter** [comment]
not our bug cannot do anything, maybe it is fixed in latest qts, i dont know
